atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java

```diff
@@ -618,6 +648,23 @@ protected void appendEntries(
     succeedAppend(lastLogIndex, future);
   }
 
+  private boolean validChecksum(final AppendRequest request, final int index) {
+    if (request.checksums() == null) {
+      return true;
+    }
+
+    final long checksum = request.checksums().get(index);
+    final RaftLogEntry entry = request.entries().get(index);
+
+    buffer.clear();
+    namespace.serialize(entry, buffer);
```

Comment:
We are serializing the entry and calculating the checksums here and again in journal::append. Can we not do that?

Reply:
I didn't see a way of doing that without refactoring the journal. We'd have to have slightly different journal implementations for the leader and follower

Reply:
Ok. Let's leave it as it is. Can you create a follow up issue for it? Did you run a benchmark with this changes - does having to serialize it twice has any impact on the performance?

Reply:
Yes, it's in the mp-4736 namespace. It's been running for 10 days without issues. The average for the last 7 days is 142 processes completed per second. I do see the checksum validation in the profiling at ~1% due to the serialization